application/common/components/adapters/SecureUserIdStore.php

```diff
@@ -0,0 +1,164 @@
+<?php
+namespace Sil\Idp\IdSync\common\components\adapters;
+
+use Exception;
+use GuzzleHttp\Client;
+use InvalidArgumentException;
+use Sil\Idp\IdSync\common\components\IdStoreBase;
+use Sil\Idp\IdSync\common\models\User;
+use yii\helpers\Json;
+
+class SecureUserIdStore extends IdStoreBase
+{
+    public $apiUrl = null;
+    public $apiKey = null;
+    public $apiSecret = null;
+
+    public $timeout = 45; // Timeout in seconds (per call to ID Store API).
+
+    protected $httpClient = null;
+
+    public function init()
+    {
+        $requiredProperties = [
+            'apiUrl',
+            'apiKey',
+            'apiSecret',
+        ];
+        foreach ($requiredProperties as $requiredProperty) {
+            if (empty($this->$requiredProperty)) {
+                throw new InvalidArgumentException(sprintf(
+                    'No %s was provided.',
+                    $requiredProperty
+                ), 1642083101);
+            }
+        }
+
+        parent::init();
+    }
+
+    public static function getFieldNameMap(): array
+    {
+        return [
+            'employee_number' => User::EMPLOYEE_ID,
+            'first_name' => User::FIRST_NAME,
+            'last_name' => User::LAST_NAME,
+            'display_name' => User::DISPLAY_NAME,
+            'email' => User::EMAIL,
+            'username' => User::USERNAME,
+            'locked' => User::LOCKED,
+            'manager_email' => User::MANAGER_EMAIL,
```

Comment:
Is this complete? I thought Schram added 2 manager fields. Just checking.

Reply:
I think Schram's addition was just to the workday store for adding the HR fields.

Reply:
Yep. Not all fields must be listed here.